src/fmu/tools/qcproperties/_propstat.py

```diff
@@ -202,6 +202,13 @@ def _codes_to_codenames(self, dframe):
                         f"{param} parameter is not!"
                     )
                 codes = xtg_prop.codes.copy()
+
+                # if no codenames, use codevalues as codenames
+                if not codes:
+                    codes = {
+                        value: str(value)
+                        for value in np.unique(xtg_prop.get_active_npvalues1d())
+                    }
```

Comment:
In case you don't want to inject "fake" dictionary values, this could perhaps alternatively be solved using the `<dict>.get()` function. If you have a dict `somedict`, you can do `somedict.get("somekey", "somekey")` which will return "somekey" if "somekey" is not in the dictionary:
```
In [1]: a = dict(foo="bar")

In [2]: a
Out[2]: {'foo': 'bar'}

In [3]: a.get("foo")
Out[3]: 'bar'

In [4]: a.get("com", "com")
Out[4]: 'com'
```

Reply:
I don't think there is a risk for fake dict values in tnatt's approach(?)

Reply:
I am not sure what you mean by "fake" values @berland ? All of the values inserted should exist in the property.

Reply:
Fake is probably an incorrect term here then ;)  Call then "undefined". The question was more if you really want to provide a modified dictionary for the default values, or if you want to code the defaulting into the code that accesses the dictionary.